src/charm.py

```diff
@@ -13,7 +13,14 @@
 from pathlib import Path
 from typing import Dict, List, Literal, Optional, Tuple, get_args
 
-import psycopg2
+try:
+    # First platform-specific import, will fail on wrong architecture
+    import psycopg2
+except ModuleNotFoundError:
+    from fake_charm import block_on_wrong_architecture
+
+    block_on_wrong_architecture()
+
```

Comment:
Q: what about to compare [platform.machine()](https://docs.python.org/3/library/platform.html#platform.machine) to the content of  `/var/lib/juju/agents/unit-opensearch-1/charm/.juju-charm` :

ch:**amd64**/jammy/opensearch-99

Reply:
Platform.machine will still return the uname architecture (`x86_64` and `aarch64`).

`.juju-charm` might be a better option, since we don't have to rely on pyyaml being available.

Reply:
`.juju-charm` for sure is a better option, wasn't aware of this, thanks for mentioning! 

About `platform.machine()` I don't really see the difference with `os.uname().machine`, same output format here.

Reply:
Sure, you can use `os.uname().machine`.
As discussed today on the sync, Marcello is willing to continue the current PR approach. ACK.